ark/phenotyping/som_utils.py

```diff
@@ -1014,64 +1335,84 @@ def cell_consensus_cluster(base_dir, max_k=20, cap=3, column_prefix='cluster',
+    ).astype(np.int64)
 
-    # verify the column prefix provided is valid
-    misc_utils.verify_in_list(
-        provided_column_prefix=column_prefix,
-        valid_column_prefixes=['cluster', 'hCluster_cap']
+    # read in the average number of pixel/SOM clusters across all cell SOM clusters
+    cell_som_cluster_avgs_and_counts = pd.read_csv(cluster_avg_path)
+
+    # merge metacluster assignments in
+    cell_som_cluster_avgs_and_counts = pd.merge_asof(
+        cell_som_cluster_avgs_and_counts, som_to_meta_data, on='cell_som_cluster'
     )
 
-    # verify the cell_cluster_col provided is valid
-    misc_utils.verify_in_list(
-        provided_cluster_col=cell_cluster_col,
-        valid_cluster_cols=['cluster', 'hCluster_cap']
+    # resave average number of pixel/SOM clusters across all cell SOM clusters
+    # with metacluster assignments
+    cell_som_cluster_avgs_and_counts.to_csv(
+        cluster_avg_path,
+        index=False
     )
 
-    # average the columns across the cluster column
-    cluster_avgs = compute_cell_cluster_count_avg(os.path.join(base_dir, cluster_name),
-                                                  column_prefix, cell_cluster_col)
+    # compute the weighted channel average expression per cell SOM cluster
+    print("Compute average weighted channel expression across cell SOM clusters")
+    cell_som_cluster_channel_avg = compute_cell_cluster_channel_avg(
+        fovs,
+        channels,
+        base_dir,
+        weighted_cell_channel_avg_name,
+        cell_consensus_name,
+        'cell_som_cluster'
+    )
 
-    # convert cluster column to integer type
-    cluster_avgs[cell_cluster_col] = cluster_avgs[cell_cluster_col].astype(int)
+    # merge metacluster assignments into cell_som_cluster_channel_avg
+    print(
+        "Mapping meta cluster values onto average weighted channel expression"
+        "across cell SOM clusters"
+    )
+    cell_som_cluster_channel_avg = pd.merge_asof(
+        cell_som_cluster_channel_avg, som_to_meta_data, on='cell_som_cluster'
+    )
 
-    # sort cluster col in ascending order
-    cluster_avgs = cluster_avgs.sort_values(by=cell_cluster_col)
-
-    # draw the heatmap
-    visualize.draw_heatmap(
-        data=cluster_avgs.drop(columns=cell_cluster_col).values,
-        x_labels=cluster_avgs[cell_cluster_col],
-        y_labels=cluster_avgs.drop(columns=cell_cluster_col).columns.values,
-        dpi=dpi, center_val=center_val, overlay_values=overlay_values,
-        colormap=colormap, save_dir=save_dir, save_file=save_file
+    # save the weighted channel average expression per cell cluster
+    cell_som_cluster_channel_avg.to_csv(
+        os.path.join(base_dir, cell_som_cluster_channel_avg_name),
+        index=False
+    )
+
+    # compute the weighted channel average expression per cell meta cluster
+    print("Compute average weighted channel expression across cell meta clusters")
+    cell_meta_cluster_channel_avg = compute_cell_cluster_channel_avg(
```

Comment:
Same thing here. Zak's tool needs un-zscored data, so this function shouldn't z-score.